cmd/clusterctl/client/repository/components.go

```diff
@@ -369,16 +386,19 @@ func addNamespaceIfMissing(objs []unstructured.Unstructured, targetNamespace str
 		}
 	}
 
-	// if there isn't an object with Kind Namespace, add it
+	// if there isn't an object with Kind Namespace, prepend it
```

Comment:
Is this introducing an assumption on ordering the consumers are relying on?
Is there a particular reason for doing so?

Reply:
> @asalkeld Welcome partying_face ! Happy to see new people joining our operator effort !
> 
> Is it possible to rebase this on top of master? this branch now is outdated and a lot of changes merged for v1alpha4 redefining how we deploy providers (e.g. there is no anymore a webhook namespace, nor instanceObj/sharedObj)

Wow, that will change things! How do you want to go about rebasing (git pull master into my PR or a separate PR)?

> 
> Also, have you considered to break down this effort in multiple PRs, since there are changes )like defining new clusterctl configProvider or new clusterctl repository providers) which seems well self contained. This will make review easier and faster...

I am happy to do that, I guess part of this PR was figuring out what needed to be done. So now that I have a good idea, we can do that. Probably a good idea to do the rebase first to see what that changes.

> 
> I still need to go into details of reconciler and component mutations, there is a lot in this PR sweat_smile

no worries, thanks for the review :smile: